src/draw/nxp/pxp/lv_pxp_osa.c

```diff
@@ -114,22 +113,32 @@ pxp_cfg_t * pxp_get_default_cfg(void)
 
 static void _pxp_interrupt_init(void)
 {
-#if defined(SDK_OS_FREE_RTOS)
-    xPXPIdleSemaphore = xSemaphoreCreateBinary();
-    PXP_ASSERT_MSG(xPXPIdleSemaphore, "xSemaphoreCreateBinary failed!");
+#if LV_USE_OS
+    if(lv_thread_sync_init(&pxp_sync) != LV_RESULT_OK) {
+        PXP_ASSERT_MSG(false, "Failed to init thread_sync.");
+    }
+#endif
 
+#if defined(SDK_OS_FREE_RTOS)
     NVIC_SetPriority(PXP_IRQ_ID, configLIBRARY_MAX_SYSCALL_INTERRUPT_PRIORITY + 1);
+    NVIC_EnableIRQ(PXP_IRQ_ID);
+#elif defined(__ZEPHYR__)
+    IRQ_CONNECT(DT_IRQN(DT_NODELABEL(pxp)), CONFIG_LV_Z_PXP_INTERRUPT_PRIORITY, PXP_ZephyrIRQHandler, NULL, 0);
```

Comment:
I understand now, you included the ZEPHYR header just for this `CONFIG_LV_Z_PXP_INTERRUPT_PRIORITY`.

We should not mix any longer this with lv_osal support.